OSDOCS-1480: Added info for region handling. (#26397)
openshift/openshift-docs: modules/moa-installing-prerequisites.adoc

====
+
. Install and configure the AWS CLI.
.. Follow the AWS documentation to install and configure the link:https://aws.amazon.com/cli/[aws-cli] for your operating system.
.. Optional: Set a default AWS region. `moactl` evaluates regions in the following priority order:
+
... The region specified when running a `moactl` command with the `--region` flag.

> ... The region set in the `AWS_DEFAULT_REGION` environment variable. See the link:https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html[AWS documentation for more details].

Reviewer: It's better to include the title of the document that you're linking to instead of a generic reference, like:

```
See link:https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html[title] in the AWS documentation.
```

Reviewer: But this is a much better way to present the information! Thank you. :)

Author: @kalexand-rh Hi, Kathryn. I updated the doc with Kevin's changes and he updated the manual preview link. So this is ready for a second round of peer review when you have a chance. Thank you!

> ... The region set in the `AWS_DEFAULT_REGION` environment variable. See the link:https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html[AWS documentation for more details].

Reviewer: ```suggestion
... The region set in the `AWS_DEFAULT_REGION` environment variable. See link:https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html[Environment variables to configure the AWS CLI] in the AWS documentation.
```

> ... The default region set in your AWS configuration file. See the link:https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html#cli-configure-quickstart-config[AWS documentation for more details].

Reviewer: ```suggestion
... The default region set in your AWS configuration file. See link:https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html#cli-configure-quickstart-config[Quick configuration with `aws configure`] in the AWS documentation.
```

Author: Yes, I understand now, I worked with Kevin on it. This has been done. I'll ask Kevin to merge the changes. Thanks for your great suggestions!

====
+
. Install and configure the AWS CLI.
.. Follow the AWS documentation to install and configure the link:https://aws.amazon.com/cli/[aws-cli] for your operating system.
+
[NOTE]
====
`moactl` Priority Order of AWS Regions::
... Option specified on the `moactl` command line +
... Environment variable +
... Default region listed in the AWS configuration file +
====

> +

Reviewer 2: This is difficult because we're trying to provide helpful pointers on how to use the AWS CLI, without getting too deep into redocumenting what AWS docs already cover.

Here's a proposed update with some links:
`. Install and configure the AWS CLI.
 .. Follow the AWS documentation to link:https://aws.amazon.com/cli/[install and configure the aws-cli] for your operating system.
 .. Optional: Set a default AWS region. `moactl` evaluates regions in the following priority order:
 +
... The region specified when running a `moactl` command with the `--region` flag.
... The region set in the `AWS_DEFAULT_REGION` environment variable. See the link:https://docs.aws.amazon.com/cli/latest/userguide/c    li-configure-envvars.html[AWS documentaiton for more details].
... The default region set in your AWS configuration file. See the link:https://docs.aws.amazon.com/cli/latest/userguide/cli-configu    re-quickstart.html#cli-configure-quickstart-config[AWS documentation for more details].`

Reviewer 2: ![image](https://user-images.githubusercontent.com/3738757/96002573-44659500-0e07-11eb-93d9-f046905c98f5.png)

+
. Install and configure the AWS CLI.
.. Follow the AWS documentation to install the link:https://aws.amazon.com/cli/[aws-cli] for your operating system.

> .. `moactl` handles AWS regions in this priority order:

Reviewer: I'm not sure where this information should go, but it should be a note somewhere else instead of a step here - there's no explicit action.
